MyApplication/app/src/main/java/test/test/com/myapplication/MainActivity.java

```diff
@@ -1,42 +1,118 @@
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-        changeFragment(new MainFragment(), MainFragment.NAME, FRAGMENT_ANIMATION);
+        ButterKnife.bind(this);
+        searchFragment = new SearchFragment();
+        searchFragment.setOnRouteReady(this);
+        mainFragment = getFragmentManager().findFragmentById(R.id.static_fragment);
+        getLocation();
+        setUpSlider();
     }
 
+    private void getLocation() {
+        if (PermissionUtils.isLocationPermissionsApproved(this)) {
+            LocationUtils.getLocation(this, new ILocationCallBack() {
+                @Override
+                public void onLocation(Location location) {
+                    if (location != null) {
+                        changeFragment(searchFragment, SearchFragment.NAME);
+                    }
+                }
+            });
+        } else {
+            PermissionUtils.requestForLocationPermissions(this);
+        }
+    }
 
-    public void changeFragment(Fragment fragment, String className, int animationState) {
-        android.support.v4.app.FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
-        int i=R.animator.slide_in_right;
-        if (animationState == FRAGMENT_ANIMATION) {
-       //    ft.setCustomAnimations(R.animator.slide_in_right, R.animator.fragment_scale_out);
-        } else if (animationState == FRAGMENT_BACK_ANIMATION) {
-       //     ft.setCustomAnimations(R.animator.fragment_scale_in, R.animator.slide_out_right);
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String permissions[], int[] grantResults) {
+        if (PermissionUtils.isLocationPermissionsApproved(this)) {
+            getLocation();
         }
+    }
+
+    private void setUpSlider() {
+        mSlider.setCoveredFadeColor(0);
+        mSlider.setOverlayed(false);
+        mSlider.setAnchorPoint(SLIDING_PANEL_ANCHOR_POINT);
+        mSlider.setPanelHeight(0);
+        mSlider.addPanelSlideListener(new SlidingUpPanelLayout.PanelSlideListener() {
+            @Override
+            public void onPanelSlide(View panel, final float slideOffset) {
+                updateMapMargins((int) (panel.getHeight() * slideOffset) / 2);
+            }
+
+            @Override
+            public void onPanelStateChanged(View panel, SlidingUpPanelLayout.PanelState previousState, SlidingUpPanelLayout.PanelState newState) {
+
+            }
+        });
+    }
+
+    private void updateMapMargins(int topMargin) {
+
+        FrameLayout.LayoutParams params = (FrameLayout.LayoutParams) mapLayout.getLayoutParams();
+        params.topMargin = -topMargin;
+        params.bottomMargin = topMargin;
+        mapLayout.setLayoutParams(params);
+    }
+
+    public void changeFragment(Fragment fragment, String className) {
+        android.support.v4.app.FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
```

Comment:
Again here - Check which fragment are you using and use one across the entire application.